code/modules/art/paintings.dm

```diff
@@ -503,6 +503,104 @@
 	result_path = /obj/structure/sign/painting
 	pixel_shift = 30
 
+/obj/item/wallframe/painting/syndicate_teleporter
+	result_path = /obj/structure/sign/painting/syndicate_teleporter
+
+/obj/structure/sign/painting/syndicate_teleporter
+	damage_deflection = 20
+	///Where a human gets teleported to once they activate the teleporter, picks from a list of GLOB.active_syndicate_beacons
+	var/atom/teleport_target
+	///If we've taken enough damage, we become unusable
+	var/integrity_compromised = FALSE
+
+/obj/structure/sign/painting/syndicate_teleporter/Initialize(mapload, dir, building)
+	. = ..()
+	RegisterSignal(src, COMSIG_ATOM_TAKE_DAMAGE, PROC_REF(on_damage_taken))
+	RegisterSignals(src, list(COMSIG_QDELETING,	COMSIG_MACHINERY_BROKEN, TELEPORTER_SET_TARGET), PROC_REF(remove_connections))
+
+/obj/structure/sign/painting/syndicate_teleporter/Destroy()
+	if(current_canvas)
+		current_canvas.forceMove(drop_location())
+	if(teleport_target)
+		teleport_target = null
+	return ..()
+
+/obj/structure/sign/painting/syndicate_teleporter/attack_hand(mob/user, list/modifiers)
+	if(!teleport_target || !ishuman(user) || !IS_TRAITOR(user) || integrity_compromised)
+		return ..()
+	var/mob/living/carbon/human/teleportee = user
+	do_teleport(teleportee, teleport_target)
+
+/obj/structure/sign/painting/syndicate_teleporter/attack_hand_secondary(mob/user, list/modifiers)
+	. = ..()
+	if(!ishuman(user) || !IS_TRAITOR(user) || integrity_compromised)
+		return SECONDARY_ATTACK_CALL_NORMAL
+
+	if(teleport_target)
+		remove_connections(src)
+		user.balloon_alert(user, "Target cleared")
+		update_appearance(UPDATE_ICON)
+		return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+
+	var/set_target = tgui_input_list(user, "Where to launch to?", "Set Teleporter?", GLOB.active_syndicate_beacons)
+	if(!set_target)
+		return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+	teleport_target = set_target
+	SEND_SIGNAL(teleport_target, PAINTING_SET_TARGET, src)
```

Comment:
Signal conventions, again. But also this is starting to not look like a good use of signals. I'll check again next review.